### ISSUE TITLE
Method to record the number of circuit evaluations when using the parameter shift rule.

### ISSUE THREAD

**Issue by Thommy257** [tensorflow/quantum#689]
Hi there,

Not a bug, but a question.

I have the following hybrid model (quantum input -> PQC -> FullyConnected neural net), which In train on some data points:

```
class HybridModel(tf.keras.models.Model):
    def __init__(self, input_layer, input_symbols, pqc):
        super(HybridModel, self).__init__()
        self.input_symbols = input_symbols
        self.input_layer = tfq.convert_to_tensor([input_layer])
        self.pqc = pqc
        self.readout_ops = [cirq.Z(q) for q in self.pqc.all_qubits()]
        self.qmodel = tfq.layers.PQC(
            self.pqc,
            self.readout_ops,
            repetitions=2048,
            initializer=tf.keras.initializers.RandomUniform(0, 1),
            differentiator=tfq.differentiators.ParameterShift()
        )
        self.fc = tf.keras.layers.Dense(1)
    
    def call(self, inputs):
        num_batch = tf.gather(tf.shape(inputs), 0)    # Get batch size: first element of shape
        # (1, 1) -> (batch, 1)
        input_circuits = tf.repeat(self.input_layer, repeats=num_batch)
        inp = tfq.resolve_parameters(input_circuits,
                       symbol_names=self.input_symbols,
                       symbol_values=tf.convert_to_tensor(inputs))
        out = self.qmodel(inp)
        out = self.fc(out)
        return tf.sigmoid(out)

model = HybridModel(some_input_layer, some_input_layer_symbols, pqc_cirq)
model.compile(loss='binary_crossentropy', optimizer=tf.keras.optimizers.Adam(lr=0.01), metrics='accuracy')
hist = model.fit(X, Y, epochs=20, batch_size=1)
```

 I would now like to record the loss over the number of total circuit evaluations (to evaluate the performance of the `ParameterShift` rule). Is there a way to do that?

Thanks in advance

**Comment by lockwo** [comment]
Probably the easiest way to do it is just to compute it based on the known information. You can just get the results of model.fit and see that each backprop (for batch_size=1) will have 2*p*rep circuit evaluations. So each epoch will have len(X) * 2 * p * rep evaluations (and the loss is given per epoch). If you are looking for more fine grained analysis, you could use `tf.GradientTape()` and step through more manually and keep track of more information. You could also write your own callback (https://www.tensorflow.org/guide/keras/custom_callback) and keep using model.fit.

**Comment by Thommy257** [comment]
Perfect, thank you very much!
